bpf/tests/pktgen.h

```diff
@@ -582,6 +714,43 @@ void pktgen__finish(const struct pktgen *builder)
 
 			break;
 
+		case PKT_LAYER_IPV6_HOP_BY_HOP:
+		case PKT_LAYER_IPV6_AUTH:
+			layer_off = builder->layer_offsets[i];
+			if (layer_off >= MAX_PACKET_OFF - sizeof(struct ipv6_opt_hdr))
+				return;
+
+			ipv6_opt_layer = ctx_data(builder->ctx) + layer_off;
+			if ((void *)(ipv6_opt_layer + 1) > ctx_data_end(builder->ctx))
+				return;
+
+			if (i + 1 >= PKT_BUILDER_LAYERS)
+				return;
+
+			switch (builder->layers[i + 1]) {
+			case PKT_LAYER_IPV6_HOP_BY_HOP:
+				ipv6_opt_layer->nexthdr = NEXTHDR_HOP;
+				break;
+			case PKT_LAYER_IPV6_AUTH:
+				ipv6_opt_layer->nexthdr = NEXTHDR_AUTH;
+			case PKT_LAYER_TCP:
```

Comment:
this is missing a `break`, no?

Reply:
Sry, I haven't tested the case of ipv6/.../auth, so the test didn't cover this part of the code.

fixed.